### PR TITLE
cooklang syntax highlighting - initial add

### PR DESCRIPTION
**Pull request by ahue**
Adds syntax highlighting for [cooklang](https://cooklang.org/docs/spec/).

### CODE REVIEW

File: components/prism-cooklang.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+						},
+						'operator': /\*%|\*|%|\|/, // already includes not yet officially implemented servings operators
+						'number': /[^{}*%]+/, // amount
+					}
+				},
+				'keyword': /[@#]/,
+			}
+		},
+		'no-quant': { // ingredient, cookware
+			'pattern': /[@#][^{@#\s]+/,
```
[the remark this answers is not in the document]
Comment (ahue) [reply components/prism-cooklang.js:56]: Currently, the language spec would allow that, so for the time being, any character can appear between # and an opening bracket is ok. If however, between an # and the next linebreak appears another # or no {, then we have a not quantified or multi token ingredient or cookware. I'll amend the tests for those edge cases and possibly change the pattern.

File: components/prism-cooklang.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+					lookbehind: true,
+					alias: 'variable',
+				},
+				'duration-group': { // {...}
+					pattern: /\{[^{}]+\}/,
+					inside: {
+						'punctuation': /[{}]/,
+						'unit': {
+							// eslint-disable-next-line regexp/no-dupe-disjunctions
+							pattern: new RegExp(/(%)/.source + timer_units),
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:124]: This wasn't a false positive from `regexp/no-dupe-disjunctions`. The regex `(%)(?:h|hours|hrs|m|min|minutes)` only matches `%h` in the string `%hours`. That's a bug and can be seen in your tests. Luckily, this can easily be fixed using a `\b`.
```suggestion
							pattern: new RegExp(/(%)/.source + timer_units + /\b/.source),
```

File: components/prism-cooklang.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			'unit': {
+				pattern: /(%)[^}]+/,
+				lookbehind: true,
+				alias: 'symbol',
+			},
+			'servings-scaler': {
+				pattern: /\*/,
+				alias: 'operator',
+			},
+			'servings-alternative-seperator': {
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:36]: Typo.

```suggestion
			'servings-alternative-separator': {
```

File: components/prism-cooklang.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		'comment': {
+			// [- comment -]
+			// -- comment
+			pattern: /\[-[\s\S]*?-\]|--.*/,
+			greedy: true,
+		},
+		'meta': { // >> key: value
+			pattern: />>.*:.*/,
+			inside: {
+				property: { // key:
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:59]: I know this is super nitpicky, but...

```suggestion
				'property': { // key:
```

File: components/prism-cooklang.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+				},
+				'amount-group': amount_group_impl
+			}
+		},
+		'timer-group': { // ~timer{...}
+			// eslint-disable-next-line regexp/sort-alternatives
+			pattern: new RegExp(/~[^{}]*\{\d+%/.source + timer_units + /\}/.source),
+			inside: {
+				'timer': {
+					pattern: /(~)[^{]+/,
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:114]: A `^` will make the regex faster while still behaving the same. The same trick can also be applied to the `cookware` and `ingredient` patterns.

```suggestion
					pattern: /(^~)[^{]+/,
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: components/prism-cooklang.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-cooklang.js]: > Currently, the language spec would allow that

What about the existing parser implementations? I looked at how other syntax highlighter handle this, and [vim](https://github.com/luizribeiro/vim-cooklang/blob/861b7abd74f4606398bb3c7bc8d326f500e1340c/syntax/cook.vim#L23) and [VSCode](https://github.com/cooklang/CookVSCode/blob/6b4a6b947d576b73a6aba85847cc68110441eedd/syntaxes/cook.tmLanguage.json#L151) only allow word characters (so roughly `\w`).

I also want to point out that the spec is fairly vague (it's not exactly a BNF grammar) and has an example that would indicate that the `.` is not part of the name:

```cook
Place the potatoes into a #pot.
Mash the potatoes with a #potato masher{}.
```

**Comment by ahue** [comment]
@RunDevelopment  - appreciate the feedback - it's really helpful!

File: components/prism-cooklang.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-cooklang.js]: Oops, my bad. There is [a grammar](https://github.com/cooklang/spec/blob/main/EBNF.md), I just didn't see it.

So, according to the grammar, only a `word` is allowed if there is no quantifier, and a `word` is `{ alphabetic character | digit | symbol character - cooklang ancillary character }-`. I don't want to bore you with reading the EBNF to you any longer, so here is the equivalent regex for `word`:

```js
/(?:(?![>|~@#:{}%])[\p{L}\p{M}0-9\p{S}])+/u
```

Unfortunetly, this uses Unicode properties and Prism still supports browsers that don't support them. So we have to compile this regex into an equivalent non-Unicode regex:

```js
/(?:[\d$+<=A-Z\x60-\x7a\xa2-\xa6\xa8-\xaa\xac\xae-\xb1\xb4\xb5\xb8\xba\xc0-\u0377\u037a-\u037d\u037f\u0384-\u0386\u0388-\u038a\u038c\u038e-\u03a1\u03a3-\u052f\u0531-\u0556\u0559\u0560-\u0588\u058d-\u058f\u0591-\u05bd\u05bf\u05c1\u05c2\u05c4\u05c5\u05c7\u05d0-\u05ea\u05ef-\u05f2\u0606-\u0608\u060b\u060e-\u061a\u0620-\u065f\u066e-\u06d3\u06d5-\u06dc\u06de-\u06ef\u06fa-\u06ff\u0710-\u074a\u074d-\u07b1\u07ca-\u07f6\u07fa\u07fd-\u082d\u0840-\u085b\u0860-\u086a\u08a0-\u08b4\u08b6-\u08c7\u08d3-\u08e1\u08e3-\u0963\u0971-\u0983\u0985-\u098c\u098f\u0990\u0993-\u09a8\u09aa-\u09b0\u09b2\u09b6-\u09b9\u09bc-\u09c4\u09c7\u09c8\u09cb-\u09ce\u09d7\u09dc\u09dd\u09df-\u09e3\u09f0-\u09f3\u09fa-\u09fc\u09fe\u0a01-\u0a03\u0a05-\u0a0a\u0a0f\u0a10\u0a13-\u0a28\u0a2a-\u0a30\u0a32\u0a33\u0a35\u0a36\u0a38\u0a39\u0a3c\u0a3e-\u0a42\u0a47\u0a48\u0a4b-\u0a4d\u0a51\u0a59-\u0a5c\u0a5e\u0a70-\u0a75\u0a81-\u0a83\u0a85-\u0a8d\u0a8f-\u0a91\u0a93-\u0aa8\u0aaa-\u0ab0\u0ab2\u0ab3\u0ab5-\u0ab9\u0abc-\u0ac5\u0ac7-\u0ac9\u0acb-\u0acd\u0ad0\u0ae0-\u0ae3\u0af1\u0af9-\u0aff\u0b01-\u0b03\u0b05-\u0b0c\u0b0f\u0b10\u0b13-\u0b28\u0b2a-\u0b30\u0b32\u0b33\u0b35-\u0b39\u0b3c-\u0b44\u0b47\u0b48\u0b4b-\u0b4d\u0b55-\u0b57\u0b5c\u0b5d\u0b5f-\u0b63\u0b70\u0b71\u0b82\u0b83\u0b85-\u0b8a\u0b8e-\u0b90\u0b92-\u0b95\u0b99\u0b9a\u0b9c\u0b9e\u0b9f\u0ba3\u0ba4\u0ba8-\u0baa\u0bae-\u0bb9\u0bbe-\u0bc2\u0bc6-\u0bc8\u0bca-\u0bcd\u0bd0\u0bd7\u0bf3-\u0bfa\u0c00-\u0c0c\u0c0e-\u0c10\u0c12-\u0c28\u0c2a-\u0c39\u0c3d-\u0c44\u0c46-\u0c48\u0c4a-\u0c4d\u0c55\u0c56\u0c58-\u0c5a\u0c60-\u0c63\u0c7f-\u0c83\u0c85-\u0c8c\u0c8e-\u0c90\u0c92-\u0ca8\u0caa-\u0cb3\u0cb5-\u0cb9\u0cbc-\u0cc4\u0cc6-\u0cc8\u0cca-\u0ccd\u0cd5\u0cd6\u0cde\u0ce0-\u0ce3\u0cf1\u0cf2\u0d00-\u0d0c\u0d0e-\u0d10\u0d12-\u0d44\u0d46-\u0d48\u0d4a-\u0d4f\u0d54-\u0d57\u0d5f-\u0d63\u0d79-\u0d7f\u0d81-\u0d83\u0d85-\u0d96\u0d9a-\u0db1\u0db3-\u0dbb\u0dbd\u0dc0-\u0dc6\u0dca\u0dcf-\u0dd4\u0dd6\u0dd8-\u0ddf\u0df2\u0df3\u0e01-\u0e3a\u0e3f-\u0e4e\u0e81\u0e82\u0e84\u0e86-\u0e8a\u0e8c-\u0ea3\u0ea5\u0ea7-\u0ebd\u0ec0-\u0ec4\u0ec6\u0ec8-\u0ecd\u0edc-\u0edf\u0f00-\u0f03\u0f13\u0f15-\u0f1f\u0f34-\u0f39\u0f3e-\u0f47\u0f49-\u0f6c\u0f71-\u0f84\u0f86-\u0f97\u0f99-\u0fbc\u0fbe-\u0fcc\u0fce\u0fcf\u0fd5-\u0fd8\u1000-\u103f\u1050-\u108f\u109a-\u10c5\u10c7\u10cd\u10d0-\u10fa\u10fc-\u1248\u124a-\u124d\u1250-\u1256\u1258\u125a-\u125d\u1260-\u1288\u128a-\u128d\u1290-\u12b0\u12b2-\u12b5\u12b8-\u12be\u12c0\u12c2-\u12c5\u12c8-\u12d6\u12d8-\u1310\u1312-\u1315\u1318-\u135a\u135d-\u135f\u1380-\u1399\u13a0-\u13f5\u13f8-\u13fd\u1401-\u166d\u166f-\u167f\u1681-\u169a\u16a0-\u16ea\u16f1-\u16f8\u1700-\u170c\u170e-\u1714\u1720-\u1734\u1740-\u1753\u1760-\u176c\u176e-\u1770\u1772\u1773\u1780-\u17d3\u17d7\u17db-\u17dd\u180b-\u180d\u1820-\u1878\u1880-\u18aa\u18b0-\u18f5\u1900-\u191e\u1920-\u192b\u1930-\u193b\u1940\u1950-\u196d\u1970-\u1974\u1980-\u19ab\u19b0-\u19c9\u19de-\u1a1b\u1a20-\u1a5e\u1a60-\u1a7c\u1a7f\u1aa7\u1ab0-\u1ac0\u1b00-\u1b4b\u1b61-\u1b7c\u1b80-\u1baf\u1bba-\u1bf3\u1c00-\u1c37\u1c4d-\u1c4f\u1c5a-\u1c7d\u1c80-\u1c88\u1c90-\u1cba\u1cbd-\u1cbf\u1cd0-\u1cd2\u1cd4-\u1cfa\u1d00-\u1df9\u1dfb-\u1f15\u1f18-\u1f1d\u1f20-\u1f45\u1f48-\u1f4d\u1f50-\u1f57\u1f59\u1f5b\u1f5d\u1f5f-\u1f7d\u1f80-\u1fb4\u1fb6-\u1fc4\u1fc6-\u1fd3\u1fd6-\u1fdb\u1fdd-\u1fef\u1ff2-\u1ff4\u1ff6-\u1ffe\u2044\u2052\u2071\u207a-\u207c\u207f\u208a-\u208c\u2090-\u209c\u20a0-\u20bf\u20d0-\u20f0\u2100-\u214f\u2183\u2184\u218a\u218b\u2190-\u2307\u230c-\u2328\u232b-\u2426\u2440-\u244a\u249c-\u24e9\u2500-\u2767\u2794-\u27c4\u27c7-\u27e5\u27f0-\u2982\u2999-\u29d7\u29dc-\u29fb\u29fe-\u2b73\u2b76-\u2b95\u2b97-\u2c2e\u2c30-\u2c5e\u2c60-\u2cf3\u2d00-\u2d25\u2d27\u2d2d\u2d30-\u2d67\u2d6f\u2d7f-\u2d96\u2da0-\u2da6\u2da8-\u2dae\u2db0-\u2db6\u2db8-\u2dbe\u2dc0-\u2dc6\u2dc8-\u2dce\u2dd0-\u2dd6\u2dd8-\u2dde\u2de0-\u2dff\u2e2f\u2e50\u2e51\u2e80-\u2e99\u2e9b-\u2ef3\u2f00-\u2fd5\u2ff0-\u2ffb\u3004-\u3006\u3012\u3013\u3020\u302a-\u302f\u3031-\u3037\u303b\u303c\u303e\u303f\u3041-\u3096\u3099-\u309f\u30a1-\u30fa\u30fc-\u30ff\u3105-\u312f\u3131-\u318e\u3190\u3191\u3196-\u31e3\u31f0-\u321e\u322a-\u3247\u3250\u3260-\u327f\u328a-\u32b0\u32c0-\u9ffc\ua000-\ua48c\ua490-\ua4c6\ua4d0-\ua4fd\ua500-\ua60c\ua610-\ua61f\ua62a\ua62b\ua640-\ua672\ua674-\ua67d\ua67f-\ua6e5\ua6f0\ua6f1\ua700-\ua7bf\ua7c2-\ua7ca\ua7f5-\ua82c\ua836-\ua839\ua840-\ua873\ua880-\ua8c5\ua8e0-\ua8f7\ua8fb\ua8fd-\ua8ff\ua90a-\ua92d\ua930-\ua953\ua960-\ua97c\ua980-\ua9c0\ua9cf\ua9e0-\ua9ef\ua9fa-\ua9fe\uaa00-\uaa36\uaa40-\uaa4d\uaa60-\uaac2\uaadb-\uaadd\uaae0-\uaaef\uaaf2-\uaaf6\uab01-\uab06\uab09-\uab0e\uab11-\uab16\uab20-\uab26\uab28-\uab2e\uab30-\uab6b\uab70-\uabea\uabec\uabed\uac00-\ud7a3\ud7b0-\ud7c6\ud7cb-\ud7fb\uf900-\ufa6d\ufa70-\ufad9\ufb00-\ufb06\ufb13-\ufb17\ufb1d-\ufb36\ufb38-\ufb3c\ufb3e\ufb40\ufb41\ufb43\ufb44\ufb46-\ufbc1\ufbd3-\ufd3d\ufd50-\ufd8f\ufd92-\ufdc7\ufdf0-\ufdfd\ufe00-\ufe0f\ufe20-\ufe2f\ufe62\ufe64-\ufe66\ufe69\ufe70-\ufe74\ufe76-\ufefc\uff04\uff0b\uff1c-\uff1e\uff21-\uff3a\uff3e\uff40-\uff5a\uff5c\uff5e\uff66-\uffbe\uffc2-\uffc7\uffca-\uffcf\uffd2-\uffd7\uffda-\uffdc\uffe0-\uffe6\uffe8-\uffee\ufffc\ufffd^]|\ud800[\udc00-\udc0b\udc0d-\udc26\udc28-\udc3a\udc3c\udc3d\udc3f-\udc4d\udc50-\udc5d\udc80-\udcfa\udd37-\udd3f\udd79-\udd89\udd8c-\udd8e\udd90-\udd9c\udda0\uddd0-\uddfd\ude80-\ude9c\udea0-\uded0\udee0\udf00-\udf1f\udf2d-\udf40\udf42-\udf49\udf50-\udf7a\udf80-\udf9d\udfa0-\udfc3\udfc8-\udfcf]|\ud801[\udc00-\udc9d\udcb0-\udcd3\udcd8-\udcfb\udd00-\udd27\udd30-\udd63\ude00-\udf36\udf40-\udf55\udf60-\udf67]|\ud802[\udc00-\udc05\udc08\udc0a-\udc35\udc37\udc38\udc3c\udc3f-\udc55\udc60-\udc78\udc80-\udc9e\udce0-\udcf2\udcf4\udcf5\udd00-\udd15\udd20-\udd39\udd80-\uddb7\uddbe\uddbf\ude00-\ude03\ude05\ude06\ude0c-\ude13\ude15-\ude17\ude19-\ude35\ude38-\ude3a\ude3f\ude60-\ude7c\ude80-\ude9c\udec0-\udee6\udf00-\udf35\udf40-\udf55\udf60-\udf72\udf80-\udf91]|\ud803[\udc00-\udc48\udc80-\udcb2\udcc0-\udcf2\udd00-\udd27\ude80-\udea9\udeab\udeac\udeb0\udeb1\udf00-\udf1c\udf27\udf30-\udf50\udfb0-\udfc4\udfe0-\udff6]|\ud804[\udc00-\udc46\udc7f-\udcba\udcd0-\udce8\udd00-\udd34\udd44-\udd47\udd50-\udd73\udd76\udd80-\uddc4\uddc9-\uddcc\uddce\uddcf\uddda\udddc\ude00-\ude11\ude13-\ude37\ude3e\ude80-\ude86\ude88\ude8a-\ude8d\ude8f-\ude9d\ude9f-\udea8\udeb0-\udeea\udf00-\udf03\udf05-\udf0c\udf0f\udf10\udf13-\udf28\udf2a-\udf30\udf32\udf33\udf35-\udf39\udf3b-\udf44\udf47\udf48\udf4b-\udf4d\udf50\udf57\udf5d-\udf63\udf66-\udf6c\udf70-\udf74]|\ud805[\udc00-\udc4a\udc5e-\udc61\udc80-\udcc5\udcc7\udd80-\uddb5\uddb8-\uddc0\uddd8-\udddd\ude00-\ude40\ude44\ude80-\udeb8\udf00-\udf1a\udf1d-\udf2b\udf3f]|\ud806[\udc00-\udc3a\udca0-\udcdf\udcff-\udd06\udd09\udd0c-\udd13\udd15\udd16\udd18-\udd35\udd37\udd38\udd3b-\udd43\udda0-\udda7\uddaa-\uddd7\uddda-\udde1\udde3\udde4\ude00-\ude3e\ude47\ude50-\ude99\ude9d\udec0-\udef8]|\ud807[\udc00-\udc08\udc0a-\udc36\udc38-\udc40\udc72-\udc8f\udc92-\udca7\udca9-\udcb6\udd00-\udd06\udd08\udd09\udd0b-\udd36\udd3a\udd3c\udd3d\udd3f-\udd47\udd60-\udd65\udd67\udd68\udd6a-\udd8e\udd90\udd91\udd93-\udd98\udee0-\udef6\udfb0\udfd5-\udff1]|\ud808[\udc00-\udf99]|\ud809[\udc80-\udd43]|[\ud80c\ud81c-\ud820\ud822\ud840-\ud868\ud86a-\ud86c\ud86f-\ud872\ud874-\ud879\ud880-\ud883][\udc00-\udfff]|\ud80d[\udc00-\udc2e]|\ud811[\udc00-\ude46]|\ud81a[\udc00-\ude38\ude40-\ude5e\uded0-\udeed\udef0-\udef4\udf00-\udf36\udf3c-\udf43\udf45\udf63-\udf77\udf7d-\udf8f]|\ud81b[\ude40-\ude7f\udf00-\udf4a\udf4f-\udf87\udf8f-\udf9f\udfe0\udfe1\udfe3\udfe4\udff0\udff1]|\ud821[\udc00-\udff7]|\ud823[\udc00-\udcd5\udd00-\udd08]|\ud82c[\udc00-\udd1e\udd50-\udd52\udd64-\udd67\udd70-\udefb]|\ud82f[\udc00-\udc6a\udc70-\udc7c\udc80-\udc88\udc90-\udc99\udc9c-\udc9e]|\ud834[\udc00-\udcf5\udd00-\udd26\udd29-\udd72\udd7b-\udde8\ude00-\ude45\udf00-\udf56]|\ud835[\udc00-\udc54\udc56-\udc9c\udc9e\udc9f\udca2\udca5\udca6\udca9-\udcac\udcae-\udcb9\udcbb\udcbd-\udcc3\udcc5-\udd05\udd07-\udd0a\udd0d-\udd14\udd16-\udd1c\udd1e-\udd39\udd3b-\udd3e\udd40-\udd44\udd46\udd4a-\udd50\udd52-\udea5\udea8-\udfcb]|\ud836[\udc00-\ude86\ude9b-\ude9f\udea1-\udeaf]|\ud838[\udc00-\udc06\udc08-\udc18\udc1b-\udc21\udc23\udc24\udc26-\udc2a\udd00-\udd2c\udd30-\udd3d\udd4e\udd4f\udec0-\udeef\udeff]|\ud83a[\udc00-\udcc4\udcd0-\udcd6\udd00-\udd4b]|\ud83b[\udcac\udcb0\udd2e\ude00-\ude03\ude05-\ude1f\ude21\ude22\ude24\ude27\ude29-\ude32\ude34-\ude37\ude39\ude3b\ude42\ude47\ude49\ude4b\ude4d-\ude4f\ude51\ude52\ude54\ude57\ude59\ude5b\ude5d\ude5f\ude61\ude62\ude64\ude67-\ude6a\ude6c-\ude72\ude74-\ude77\ude79-\ude7c\ude7e\ude80-\ude89\ude8b-\ude9b\udea1-\udea3\udea5-\udea9\udeab-\udebb\udef0\udef1]|\ud83c[\udc00-\udc2b\udc30-\udc93\udca0-\udcae\udcb1-\udcbf\udcc1-\udccf\udcd1-\udcf5\udd0d-\uddad\udde6-\ude02\ude10-\ude3b\ude40-\ude48\ude50\ude51\ude60-\ude65\udf00-\udfff]|\ud83d[\udc00-\uded7\udee0-\udeec\udef0-\udefc\udf00-\udf73\udf80-\udfd8\udfe0-\udfeb]|\ud83e[\udc00-\udc0b\udc10-\udc47\udc50-\udc59\udc60-\udc87\udc90-\udcad\udcb0\udcb1\udd00-\udd78\udd7a-\uddcb\uddcd-\ude53\ude60-\ude6d\ude70-\ude74\ude78-\ude7a\ude80-\ude86\ude90-\udea8\udeb0-\udeb6\udec0-\udec2\uded0-\uded6\udf00-\udf92\udf94-\udfca]|\ud869[\udc00-\udedd\udf00-\udfff]|\ud86d[\udc00-\udf34\udf40-\udfff]|\ud86e[\udc00-\udc1d\udc20-\udfff]|\ud873[\udc00-\udea1\udeb0-\udfff]|\ud87a[\udc00-\udfe0]|\ud87e[\udc00-\ude1d]|\ud884[\udc00-\udf4a]|\udb40[\udd00-\uddef])+/
```

Oof. That's a mouthful.

I think it might be best to very roughly approximate this regex as "`/[\d$+<=A-Z]/` and all non-ASCII characters that are not spaces." Using this approximation, we get:

```js
/(?:(?!\s)[\d$+<=A-Z\x80-\uFFFF])+/
```

Please use this regex for `single_token_suffix`.

[the remark this answers is not in the document]
Comment (ahue) [reply components/prism-cooklang.js]: Is it possible that you missed the lower case characters? 

Using the suggested one breaks the highlighting for `@milk` while it continues to work for `@MILK`.

`/(?:(?!\s)[\d$+<=a-zA-Z\x80-\uFFFF])+/`

[the remark this answers is not in the document]
Comment (RunDevelopment) [reply components/prism-cooklang.js]: Ah, yes, I did. Your regex is correct.

File: components/prism-cooklang.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+				'ingredient-keyword': {
+					pattern: /^@/,
+					alias: 'keyword',
+				},
+				'amount-group': amount_group_impl,
+			}
+		},
+		'timer-group': { // ~timer{...}
+			// eslint-disable-next-line regexp/sort-alternatives
+			pattern: new RegExp(/~[^{}]*\{\d+%/.source + timer_units + /\}/.source),
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:122]: 1. The EBNF also allows spaces between the amount/duration tokens. It might be better to be a bit more lenient here and to allow pretty much anything inside the braces. Being more lenient also has the benefit that we will give good syntax highlighting while typing.
2. I think `~[^{}]*` is a little too general. We should probably exclude a few of Cooklang's special characters.
```suggestion
			pattern: /~(?!\s)[^@#~{}]*\{[^{}]*\}/,
```

File: components/prism-cooklang.js [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+	var amount_group_impl = {
+		pattern: new RegExp(/\{/.source
+			+ '(?:(?:'
+			+ /(?:[^{}|*%]+\*?)/.source // optional serving scale
+			+ '|'
+			+ /(?:(?:[^{}|*%]+\|)+[^{}|*%]+)/.source // serving alternatives
+			+ ')' + /(?:%[^{}|*%]+)?/.source //optional unit
+			+ ')?'
+			+ /\}/.source
+		),
```
Comment (RunDevelopment) [inline components/prism-cooklang.js:20]: This pattern should not be so strict. The `multi_token_suffix` pattern allows a lot, so this pattern being so strict will cause what you can see in `timer_feature.test`: 

```json
	["ingredient-group", [
		["ingredient-keyword", "@"],
		["ingredient", "milk"],
		"{3%*%L}"
	]],
```

In general, it's better to make the parent pattern (in this case `multi_token_suffix`) stricter than the inside pattern (in this case `amount_group_impl`). 

However, I think that it might better to just allow pretty much anything inside braces. Prism isn't a parser. We can highlight syntactically incorrect code however we want. We typically even want to highlight some invalid code to give a better experience while typing.

```suggestion
		pattern: /\{[^{}]*\}/,
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

**Review by RunDevelopment — APPROVED** [review]

**Comment by RunDevelopment** [comment]
Thank you for contributing @ahue!

**Comment by ahue** [comment]
Thanks for running this project!
